synapseclient/api/wiki_service.py

```diff
@@ -0,0 +1,464 @@
+"""This module is responsible for exposing the services defined at:
+<https://rest-docs.synapse.org/rest/#org.sagebionetworks.repo.web.controller.WikiController2>
+"""
+
+import json
+from typing import TYPE_CHECKING, Any, AsyncGenerator, Dict, List, Optional
+
+if TYPE_CHECKING:
+    from synapseclient import Synapse
```

Comment:
@BryanFauble something to consider when working on the new functional interface, is that currently given this import of `Synapse` class, there will be a circular dependency if we switched `Synapse.getWiki` to use this service.

Reply:
By putting this behind the `if TYPE_CHECKING` it won't be a circular dependency. This is why we have a bunch of the the lazy imports in the code like:

```
    from synapseclient import Synapse

    client = Synapse.get_client(synapse_client=synapse_client)
```

I agree that it's unfortunate to have to deal with this issue, but there are ways to work around it.